TotalCrossVM/CMakeLists.txt

```
@@ -0,0 +1,620 @@
+cmake_minimum_required(VERSION 3.13.5)
+
+# set the project name
+project(TCVM VERSION 1.0)
+
+if(NOT DEFINED BUILD_SHARED_LIBS AND NOT DEFINED TCVM_SHARED)
+  if(CMAKE_GENERATOR STREQUAL Xcode)
+    set(TCVM_SHARED OFF) 
+    message("tcvm: ${TCVM_SHARED}")  
+  else() # others
+    set(TCVM_SHARED ON) 
+  endif()
+endif()
+
+# Find SDL2 and Skia
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/modules")
+
+# Find SDL2
+IF (NOT (DEFINED ANDROID_ABI OR MSVC OR CMAKE_GENERATOR STREQUAL Xcode))
+  find_package(SDL2 REQUIRED)
+  include_directories(${SDL2_INCLUDE_DIR})
+ENDIF (NOT (DEFINED ANDROID_ABI OR MSVC OR CMAKE_GENERATOR STREQUAL Xcode))
+
+# Find Skia
+if(NOT MSVC)
+  find_package(Skia REQUIRED)
+  include_directories(${SKIA_INCLUDE_DIRS})
+endif(NOT MSVC)
+
+# Set C/CXX flags
+if(UNIX)
+  set(UNIX_C_FLAGS "-Wno-everything")
+  set(UNIX_CXX_FLAGS "-Wno-everything")
+endif(UNIX)
+
+if(APPLE)
+  set(MACOS_C_FLAGS "-mmacosx-version-min=10.11")
+  set(MACOS_CXX_FLAGS "-mmacosx-version-min=10.11")
```

I think it makes it easier to find where we are changing those variables instead of adding an indirection for it.

```suggestion
  set(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -mmacosx-version-min=10.11")
  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -mmacosx-version-min=10.11")
```

---

We were thinking about code reuse, but as it is a small snippet and as we will not have many flags, it can really make reading easier. @flsobral what do you think about it?

---

I agree, indirections are bad for readability.

---

I changed the code to use generator expressions and modern cmake compile features. @acmlira is not a big fan of generator expressions, what do you think @otavio ?

---

fixed